compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -4758,32 +4567,12 @@ genericLoad(TR::Node * node, TR::CodeGenerator * cg, TR::MemoryReference * tempM
 
 
 /**
- * This monster function handles all loads on zos and there are many, many variations. If we had a guarantee that we are executing
- * on GoldenEagle or better, this function would had been just couple of lines to pick the right instruction of the loadInstr
- * array(!!!):
```

Comment:
Finally! I've gone ahead and did this and indeed this template function is very easy to implement.